pkg/compactor/job_wait_period.go

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package compactor
+
+import (
+	"context"
+	"path"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/thanos-io/objstore"
+
+	"github.com/grafana/mimir/pkg/storage/tsdb/block"
+)
+
+// jobWaitPeriodElapsed returns whether the 1st level compaction wait period has
+// elapsed for the input job.
+func jobWaitPeriodElapsed(ctx context.Context, job *Job, waitPeriod time.Duration, userBucket objstore.Bucket) (bool, error) {
+	if waitPeriod <= 0 {
+		return true, nil
+	}
+
+	if job.MinCompactionLevel() > 1 {
+		return true, nil
+	}
+
+	if ok, err := jobContainsBlocksUploadedAfter(ctx, job, userBucket, time.Now().Add(-waitPeriod)); err != nil {
+		return false, err
+	} else if ok {
+		return false, nil
+	}
+
+	return true, nil
```

Comment:
```suggestion
	return jobContainsBlocksUploadedAfter(ctx, job, userBucket, time.Now().Add(-waitPeriod))
```

After this change, `jobWaitPeriodElapsed` is too shallow to be separate from `jobContainsBlocksUploadedAfter`.

Reply:
Logically it's not the same. When `jobContainsBlocksUploadedAfter()` returns true, `jobWaitPeriodElapsed()` should return false. Anyway, I think we can just merge the two functions.